app/controllers/new-signup.js

```diff
@@ -0,0 +1,104 @@
+import Controller from '@ember/controller';
+import { action, set } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import registerUser from '../utils/register-api';
+import { GOTO_URL } from '../constants/signup';
+import { NEW_SIGNUP_FLOW } from '../constants/analytics';
+import { ERROR_MESSAGES, NEW_SIGNUP_STEPS } from '../constants/new-signup';
+import checkUserName from '../utils/check-username';
+
+export default class NewSignUpController extends Controller {
+  @service analytics;
+
+  queryParams = ['currentStep'];
+
+  @tracked isLoading = false;
+  @tracked isButtonDisabled = true;
+  @tracked error = '';
+  @tracked currentStep = NEW_SIGNUP_STEPS[0];
+  FIRST_STEP = NEW_SIGNUP_STEPS[0];
+  SECOND_STEP = NEW_SIGNUP_STEPS[1];
+  THIRD_STEP = NEW_SIGNUP_STEPS[2];
+  FOURTH_STEP = NEW_SIGNUP_STEPS[3];
+  LAST_STEP = NEW_SIGNUP_STEPS[4];
+
+  @tracked signupDetails = {
+    firstName: '',
+    lastName: '',
+    username: '',
+  };
+
+  @action firstStep() {
+    this.currentStep = this.SECOND_STEP;
+    this.analytics.trackEvent(NEW_SIGNUP_FLOW.USER_GETTING_STARTED);
+  }
+
+  @action secondStep() {
+    this.currentStep = this.THIRD_STEP;
+    this.analytics.trackEvent(NEW_SIGNUP_FLOW.USER_FIRST_NAME);
+    this.isButtonDisabled = true;
+  }
+
+  @action thirdStep() {
+    this.currentStep = this.FOURTH_STEP;
+    this.analytics.trackEvent(NEW_SIGNUP_FLOW.USER_LAST_NAME);
+    this.isButtonDisabled = true;
+  }
+
+  @action fourthStep() {
+    this.analytics.trackEvent(NEW_SIGNUP_FLOW.USER_USERNAME);
+    this.isButtonDisabled = true;
+    this.signup();
+  }
+
+  @action lastStep() {
```

Comment:
Can we please name the actions better according to their usage ?

Reply:
yes we can, thanks, I will have a look at it